src/sys/derivs.cxx

```diff
@@ -417,21 +414,28 @@ Field3D VDDY(const Field3D &v, const Field3D &f, CELL_LOC outloc, const std::str
 ////////////// Z DERIVATIVE /////////////////
 
 // special case where both are 2D
-Field2D VDDZ(const Field2D &UNUSED(v), const Field2D &f, CELL_LOC outloc, const
-    std::string &UNUSED(method), const std::string& UNUSED(region)) {
-  if (outloc == CELL_DEFAULT) {
-    outloc = f.getLocation();
-  }
-  return zeroFrom(f).setLocation(outloc);
+Coordinates::metric_field_type VDDZ(const Field2D &v, const Field2D &f,
+                                    CELL_LOC outloc, const std::string &method,
+                                    const std::string& region) {
+  return bout::derivatives::index::VDDZ(v, f, outloc, method, region)
+         / f.getCoordinates(outloc)->dz;
 }
 
 // Note that this is zero because no compression is included
-Field2D VDDZ(const Field3D &UNUSED(v), const Field2D &f, CELL_LOC outloc, const
-    std::string &UNUSED(method), const std::string& UNUSED(region)) {
+Coordinates::metric_field_type VDDZ(const Field3D &v, const Field2D& f,
+                                    CELL_LOC outloc, const std::string& method,
+                                    const std::string& region) {
+  // Should we take location from v or f?
```

Comment:
Needs answering?

Reply:
I think the `// Should we take location from v or f?` comment is outdated and can just be deleted. `bout::derivatives::index::VDDZ()`, etc., should handle the locations correctly.

> Just noted we use quite frequently outloc, without checking it is CELL_DEFAULT to get the coordinates object. I guess that is wrong? Should I update them all?

`f.getCoordinates(CELL_DEFAULT)` will get the `Coordinates` for `f`'s location, so the current code should be correct. It is an error to use `mesh->getCoordinates(CELL_DEFAULT)` - hopefully we don't have any of those, but if you see somewhere we call `mesh->getCoordinates(outloc)` without handling `CELL_DEFAULT` first, then that needs fixing.